chain/network/src/network_protocol/mod.rs

```diff
@@ -0,0 +1,189 @@
+use near_primitives::network::PeerId;
+use near_primitives::syncing::{EpochSyncFinalizationResponse, EpochSyncResponse};
+use near_primitives::transaction::SignedTransaction;
+use near_primitives::types::{EpochId, ProtocolVersion};
+use near_primitives::version::PEER_MIN_ALLOWED_PROTOCOL_VERSION;
+use std::fmt;
+
+#[derive(PartialEq, Eq, Clone, Debug)]
+/// Structure representing handshake between peers.
+pub struct Handshake {
+    /// Current protocol version.
+    pub(crate) protocol_version: u32,
+    /// Oldest supported protocol version.
+    pub(crate) oldest_supported_version: u32,
+    /// Sender's peer id.
+    pub(crate) sender_peer_id: PeerId,
+    /// Receiver's peer id.
+    pub(crate) target_peer_id: PeerId,
+    /// Sender's listening addr.
+    pub(crate) sender_listen_port: Option<u16>,
+    /// Peer's chain information.
+    pub(crate) sender_chain_info: PeerChainInfoV2,
+    /// Represents new `edge`. Contains only `none` and `Signature` from the sender.
+    pub(crate) partial_edge_info: PartialEdgeInfo,
+}
+
+impl Handshake {
+    pub(crate) fn new(
+        version: ProtocolVersion,
+        peer_id: PeerId,
+        target_peer_id: PeerId,
+        listen_port: Option<u16>,
+        chain_info: PeerChainInfoV2,
+        partial_edge_info: PartialEdgeInfo,
+    ) -> Self {
+        Handshake {
+            protocol_version: version,
+            oldest_supported_version: PEER_MIN_ALLOWED_PROTOCOL_VERSION,
+            sender_peer_id: peer_id,
+            target_peer_id,
+            sender_listen_port: listen_port,
+            sender_chain_info: chain_info,
+            partial_edge_info,
+        }
+    }
+}
+
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub enum HandshakeFailureReason {
+    ProtocolVersionMismatch { version: u32, oldest_supported_version: u32 },
+    GenesisMismatch(GenesisId),
+    InvalidTarget,
+}
+
+#[derive(PartialEq, Eq, Clone, Debug, strum::AsStaticStr, strum::EnumVariantNames)]
+#[allow(clippy::large_enum_variant)]
+pub enum PeerMessage {
+    Handshake(Handshake),
+    HandshakeFailure(PeerInfo, HandshakeFailureReason),
+    /// When a failed nonce is used by some peer, this message is sent back as evidence.
+    LastEdge(Edge),
+    /// Contains accounts and edge information.
+    SyncRoutingTable(RoutingTableUpdate),
+    RequestUpdateNonce(PartialEdgeInfo),
+    ResponseUpdateNonce(Edge),
+
+    PeersRequest,
+    PeersResponse(Vec<PeerInfo>),
+
+    BlockHeadersRequest(Vec<CryptoHash>),
+    BlockHeaders(Vec<BlockHeader>),
+
+    BlockRequest(CryptoHash),
+    Block(Block),
+
+    Transaction(SignedTransaction),
+    Routed(Box<RoutedMessage>),
+
+    /// Gracefully disconnect from other peer.
+    Disconnect,
+    Challenge(Challenge),
+    EpochSyncRequest(EpochId),
+    EpochSyncResponse(Box<EpochSyncResponse>),
+    EpochSyncFinalizationRequest(EpochId),
+    EpochSyncFinalizationResponse(Box<EpochSyncFinalizationResponse>),
+
+    RoutingTableSyncV2(RoutingSyncV2),
```

Comment:
Can we drop V2 from here?

Reply:
This thing is unused, the current plan is to get rid of it together with the whole RoutingTable. I'd prefer not to touch it for now.